enhancements/network/routed-ingress-support-for-primary-udn-attached-vms-with-static-ips.md

```diff
@@ -0,0 +1,742 @@
+use-case) is outside the scope for this feature.
+
+## Proposal
+
+This proposal to import VMs from other virtualization platforms while
+preserving their networking configuration will impact three different OpenShift
+components:
+- MTV
+- KubeVirt's ipam-extensions
+- OVN-Kubernetes
+
+We will elaborate the overall flow first, before digging into details in each
+of the components. There will be two sub-sections: one about preserving the
+[IP addresses](#preserving-the-original-vm-ip-address),
+another about preserving the
+[gateway configuration](#preserving-the-vm-gateway).
+MTV already introspects the VM on the source cluster, and templates the VM with
+its original MAC addresses.
+
+### Current limitations
+
+- MTV does **not** support importing VMs with IPv6 addresses. Whatever support
+  for IPv6 we come up with, will probably be limited by this.
+
+### Preserving the original VM IP address
+
+MTV currently introspects the VM to learn the MAC addresses of the interfaces,
+and templates the VM which will be created in OpenShift Virtualization with the
+required MAC address. We would need MTV to somehow also figure out what IP
+addresses are on the aforementioned interfaces.
+
+**NOTE:** a separate component can introspect the VMs in their source cluster
+and identify their IPs. It really does **not** need to be MTV to do it.
+
+For doing this, we need to have a MAC to IP addresses association stored in the
+cluster (*at least* for the VMs which will be migrated into OpenShift Virt).
+There are two alternatives to persist this information: a
+[centralized approach](#centralized-ip-management),
+which has the benefits of a single centralized place an admin type of user
+could provision / get information from, or a
+[de-centralized approach](#de-centralized-ip-management), similar
+to the existing `IPAMClaim` CRD, where each VM connection to a primary UDN
+information would be recorded.
+
+Independently of which option we take, the API between OpenShift Virt (CNV) and
+OVN-Kubernetes will be the same, and is roughly described in
+[CNV to OVN-Kubernetes API](#cnv-ovnk-api).
+
+#### Centralized IP management
+
+A new CRD - named `IPPool`, or `DHCPLeaseConfig` (or the like) - will be
+created, and is associated to a cluster UDN. This CRD holds the association of
+MAC address to IPs for a UDN. When importing the VM into OpenShift Virt, MTV
+(or a separate, dedicated component) will provision / update this object with
+the required information (MAC to IPs association).
+This object is providing to the admin user a single place to check the IP
+address MAC to IPs mapping. On an first implementation phase, we can have the
+admin provision these CRs manually. Later on, MTV (or any other cluster
+introspection tool) can provision these on behalf of the admin.
+
+This approach has the following advantages:
+- single place the admin to manage for UDN
+- similar to what is done on VmWare
+- simple association between the `IPPool` and the logical network
```

Comment:
I would also add a Pro: that it would be easier to spot nasty duplicate IPs that can cause unintended traffic behavior.
(not saying I prefer this approach of course)